.dev/revdep_compare_releases.R

```diff
@@ -0,0 +1,246 @@
+        )
+      },
+      error = function(cond) {
+        writeLines(conditionMessage(cond), file.path(out_dir, paste0(package, ".failure")))
+      }
+    ),
+    warning = function(cond) {
+      warnings <<- c(warnings, conditionMessage(cond))
+      invokeRestart("muffleWarning")
+    }
+  )
+  if (length(warnings) > 0L) {
+    writeLines(warnings, file.path(out_dir, paste0(package, ".warnings")))
+  }
+}
+
+load_lints <- function(version, filter_errors = TRUE) {
+  csv_files <- list.files(result_path(version), pattern = "\\.csv$", full.names = TRUE)
+  names(csv_files) <- gsub("\\.csv$", "", basename(csv_files))
+  lints <- data.table::rbindlist(lapply(csv_files, utils::read.csv), idcol = "repo")
+  if (filter_errors) lints <- lints[, if (!any(type == "error")) .SD, by = .(repo, filename)]
+  lints
+}
+
+as.data.table.proc_time <- function(x) {
+  data.table(user = x[[1L]], system = x[[2L]], elapsed = x[[3L]])
+}
+
+match_and_strip <- function(x, pattern) gsub(pattern, "", grep(pattern, x, value = TRUE))
+
+summarize_failures <- function(version, failures) {
+  files <- result_path(version, failures)
+  packages <- gsub("\\.failures$", "", failures)
+
+  package_failures <- sapply(files, function(x) paste(unique(readLines(x)), collapse = " ||| "))
+
+  paste(sprintf("  %s: %s", packages, package_failures), collapse = "\n")
+}
+
+summarize_lint_delta <- function(new, old) {
+  new_version <- new$version[1L]
+  old_version <- old$version[1L]
+
+  new_error_files <- new[type == "error", .(. = TRUE), by = .(package, filename)]
+  old_error_files <- new[type == "error", .(. = TRUE), by = .(package, filename)]
+
+  all_error_files <- merge(
+    new_error_files, old_error_files,
+    by = c("package", "filename"), suffixes = c("new", "old"),
+    all = TRUE
+  )
+  if (anyNA(all_error_files)) {
+    if (anyNA(all_error_files$.new)) {
+      all_error_files[
+        is.na(.new),
+        message("Linting failed ", old_version, " but not ", new_version, ": ", toString(filename))
+      ]
+    }
+    if (anyNA(all_error_files$.old)) {
+      all_error_files[
+        is.na(.old),
+        message("Linting failed ", new_version, " but not ", old_version, ": ", toString(filename))
+      ]
+    }
+  }
+  new <- new[!all_error_files, on = c("package", "filename")]
+  old <- old[!all_error_files, on = c("package", "filename")]
+
+  new_only <- new[!old, on = c("package", "filename", "line_number")]
+  new_only[, version := NULL]
+
+  old_only <- old[!new, on = c("package", "filename", "line_number")]
+  old_only[, version := NULL]
+
+  withr::local_options(datatable.print.nrows = Inf)
+
+  message("Count of lints found on ", new_version, " but not on ", old_version, ": ", nrow(new_only))
+  message("Count of these by linter:")
+  print(new_only[, .N, by = linter][order(-N)])
+  message("Sample of <=10 hits from each linter:")
+  print(new_only[sample(.N), head(.SD, 10L), by = linter][, .(package, filename, linter, line)])
```

Comment:
this is a bit verbose.
Maybe also reorder to `linter, line, package, filename`?
The actual line of code seems more important to assess accuracy.

Reply:
this is mainly for our own inspection, locally, so I'm not so worried about verbosity, though the column ordering can be improved.

the next step is to produce e-mails with the delta for each individual package -- there we'll be more careful about presentation.

Reply:
column order didn't matter too much -- the issue is there are v. wide lines that always bump the other columns. so I combined `line [linter]` as one column to force them always to appear together (ditto for `package:filename`)